Add PHP 8.1 Support

gnat42 opened the pull request:
This adds supports for PHP 8.1

gnat42 commented:
This is based on top of the PR for 8.0 rebased on top of the latest upstream master

rcdevs commented:
The patch is incomplete and does not handle params by reference: 

> in ./zend/callable.h 
> the types' assignment code like:

  case Type::String:      info->type = (zend_type) ZEND_TYPE_INIT_CODE(IS_STRING, arg.allowNull(), 0);

should be:

  case Type::String:      info->type = (zend_type) ZEND_TYPE_INIT_CODE(IS_STRING, arg.allowNull(), _ZEND_ARG_INFO_FLAGS(arg.byReference(), 0, 0));

gnat42 commented:
Nice catch. I've added the code (to more than just the string type parameter) I also found a php8.2 PR in a fork of this that also fixed another issue with my original php80 PR. Both of changes are in the original PHP8.0 PR since that's where it was changed and the original PR hasn't been merged.
